### PR TITLE
Charging is delayed for better pricing options

### PR DESCRIPTION
**Pull request by GergKllai1**
# PT Link

[Charging is delayed for better pricing options](https://www.pivotaltracker.com/story/show/163740872)

# User Story

```
As a user
In order to gain the most beneficial pricing option
I would like the charge to my credit card to delayed
```

# Description of PR

- Adds delayed_jobs for delaying charge
- Adds delayed_job initializer for running the jobs in the background
- Test feature using TimeCop
- Refactors transaction controller extracts code to CreditCardService
- Connects transaction to session

### CODE REVIEW

File: app/models/price_table.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class PriceTable < ApplicationRecord
-    validates_presence_of :trainees, :pricePoint
+    validates_presence_of :trainees, :price_point
     has_many :sessions
-    enum pricePoint: { low: 0, medium: 1, high: 2 }
+    enum price_point: { low: 0, medium: 1, high: 2 }
```
Comment (HannaSwDn) [inline app/models/price_table.rb:4]: Why did you change the variable name to snake case again?

File: app/models/session.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class Session < ApplicationRecord
-	validates_presence_of :title, :startDate
+	validates_presence_of :title, :start_date
```
Comment (HannaSwDn) [inline app/models/session.rb:2]: Same here, why did you change it back to snake case?

File: app/models/price_table.rb (hunk 1 above)
Comment (GergKllai1) [reply app/models/price_table.rb:4]: It's  a rails db column, rails convention to use snake case as we did with every other db column in our db. 
Also the tests were failing

File: app/models/session.rb (hunk 2 above)
Comment (GergKllai1) [reply app/models/session.rb:2]: Same above

File: app/models/price_table.rb (hunk 1 above)
Comment (HannaSwDn) [reply app/models/price_table.rb:4]: Okay I see!

File: config/puma.rb [hunk 3]
```diff
@@ -1,34 +1,15 @@
...
-# process behavior so workers use less memory.
-#
-# preload_app!
-
-# Allow puma to be restarted by `rails restart` command.
-plugin :tmp_restart
+on_worker_boot do
+  # Worker specific setup for Rails 4.1+
+  # See: https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#on-worker-boot
+  ActiveRecord::Base.establish_connection
```
Comment (johannaanden) [inline config/puma.rb]: Remove comments

File: lib/tasks/kill_postgres_connections.rake [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  | grep postgres: \
+  | grep #{db_name} \
+  | grep -v grep \
+  | awk '{print $1}' \
+  | xargs kill
+EOF
+  puts `#{sh}`
+end
+
+task "db:drop" => :kill_postgres_connections
```
Comment (johannaanden) [inline lib/tasks/kill_postgres_connections.rake:14]: What does this file do? And what do we need it for?

**Review by johannaanden — CHANGES_REQUESTED** [review]

File: lib/tasks/kill_postgres_connections.rake (hunk 4 above)
Comment (GergKllai1) [reply lib/tasks/kill_postgres_connections.rake:14]: The active jobs are running in the background, and using the database, so if we want to drop it we need to kill them.
This script does that

File: app/services/credit_card_service.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+require 'stripe'
+
+class CreditCardService
+
+	def self.create_stripe_customer(current_user,params)
+		Stripe::Customer.create(
+			email: current_user.email,
+			source: get_token(params),
+			description: [current_user.first_name, current_user.last_name].join('')
```
Comment (magnus-thor) [inline app/services/credit_card_service.rb]: this will join the first and last name together without a space in between the names, is that intended behavior?

Comment (GergKllai1) [reply app/services/credit_card_service.rb]: Nope, corrected!
